### DISCUSSION TITLE
Limiting Fields from getFullList

### DISCUSSION THREAD

**Discussion by fozzedout** [pocketbase/pocketbase#2994] [Q&A] [answered]
Using node.js to connect to Pocketbase, and I am wanting to just return 2 fields in the getFullList instead of everything.
No problem, there is an option for this. And it works. Sort of.
```
    const resultList = await pb
                        .collection('progression')
                        .getFullList( { 
                            filter: `created >= '2023-07-25'`, 
                            fields: `progression_type,logged`  } );

    console.log(resultList);
```
This then outputs this data:
```
[
  Record {
    logged: '2023-07-25 14:59:32.000Z',
    progression_type: 'mam70bvxd5fj4np',
    id: '',
    created: '',
    updated: '',
    collectionId: '',
    collectionName: '',
    expand: {}
  }
]
```
While it didn't return things like the id, collectionId, etc, the fields are still there as the return type is Record... but I don't want to return the entire record, I just want the 2 fields.

Currently, I have a loop that pushes the 2 fields into a new array of just the two fields:
```
    /**
     * @typedef {Object} limited
     * @property {string} progression_type
     * @property {string} logged
     */
    /** @type {limited[]} */
    let results = [];

    resultList.forEach(result => {
        results.push({progression_type: result.progression_type, logged: result.logged})
    });

    console.log(results);
```
so I get this result:
```
[
  {
    progression_type: 'mam70bvxd5fj4np',
    logged: '2023-07-25 14:59:32.000Z'
  }
]
```
While this works, it just seems like I'm missing something and I'm adding in extra steps to get the output I want from the db.
Is there a way to restrict or change the output of the getFullList to only include the requested fields, rather than returning the Record type?

**Comment by ganigeorgiev** [comment, marked as answer]
I think this is the same as https://github.com/pocketbase/pocketbase/discussions/2925.

What you are seeing is the serialization of the JS SDK [Record DTO model](https://github.com/pocketbase/js-sdk/blob/master/src/models/Record.ts) and not of the actual fields that are returned in the JSON response.

With the planned JS SDK refactoring after v0.17.0, the custom DTO classes may get removed to avoid also the need of explicit `structuredClone` call when moving/serializing data from one context to another (_often required when working with SSR_) and you'll receive a POJO with just the fields that are returned.
